programs/stake/src/stake_state.rs

```diff
@@ -488,6 +489,24 @@ impl Authorized {
                 self.staker = *new_authorized
             }
             StakeAuthorize::Withdrawer => {
+                if let Some((lockup, clock, custodian)) = lockup_custodian_args {
+                    if lockup.is_in_force(&clock, None) {
```

Comment:
I totally reworked this + tests.  This is much more verbose than the `withdraw` custodian check, which only returns `LockupInForce` for all these error cases, but generally much more helpful for the user.   

The only slightly confusing error case is when the custodian signed but doesn't match the lockup and `LockupInForce` is returned instead of an error code like `InvalidCustodian`.  I decided against this error code because then I'd need to replicate some of the logic in `is_in_force()` here and I'd rather not have that code in multiple places

Reply:
Ah yeah, the short-circuit...

> The only slightly confusing error case is...

I think that's fine.

An alternative to the dupe would be to replace `Lockup::is_in_force() -> bool` with `Lockup::permits() -> Result<(),StakeError>` and get the improved errors in withdraw as well.  Getting a bit out of scope though

Reply:
Agree